src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs

```diff
@@ -316,10 +316,51 @@ public static TypeDescriptor getTypeDescriptor(IntPtr thisHandle, CORINFO_CLASS_
                 hasSignificantPadding = ecmaType.IsExplicitLayout || ecmaType.GetClassLayout().Size > 0;
             };
 
-            uint fieldCount = 0;
+            // Replicate the logic in \ILToLLVMImporter.cs GetLLVMTypeForTypeDesc but without Linq
+            // Build the TypeDescriptor with the largest fields at each offset,
+            // removing overlapping fields - i.e. union or explicit layouts
```

Comment:
So...

Given we now have this logic in the VM instead of the Jit, should the same logic in the Jit be deleted?

Reply:
Honestly I forget we had the same logic in `getStructDesc` .  I'll see why it didn't work for `Register` with a view to removing it from the VM.

Reply:
I had 2 bugs in the original Jit code.   I was using the field's owning type for the class handle, not the actual field, and I was not clearing out the covered fields in union.  Having addressed those, then the `Register` struct generated correctly.